x/auth/migrations/migrations.go

```diff
@@ -0,0 +1,177 @@
+package migrations
```

Comment:
Let's please keep consistency with other package's migration scripts:

- keeper has a `migrations.go` file, inside there's a Migrator and its Migrate1to2 function.
- all migration logic itself should go inside `legacy/v043/store.go` (the `v043` subfolder is the current way of doing folder-level versioning)

Please see x/staking, x/bank, x/distribution or x/gov for examples.

FTR, I actually prefer the `migrations/` folder name instead of `legacy/`, and have proposed exactly this once or twice, but for now I prefer consistency. If you have some ideas, let's maybe create an issue after this PR to rethink migrations folder structure.

Reply:
I moved it out to fix an import cycle.

Reply:
I encountered it too.

We can get around it by declaring a local interface with the methods we need, but it's an ugly hack...

Reply:
Refactored to fix import cycle while still allowing for @amaurym suggestions.

Had to add some logic in the `Migrate1to2` method to do so, but the trade-off looks okay to me.